_posts/2023-04-21-tensorflow-mnist-classification.adoc

```diff
@@ -0,0 +1,405 @@
+---
+layout: post
+title:  Image classification with Debezium and TensorFlow
+date:   2023-04-21 10:20:00 -0100
+tags: [ machine-learning, tensorflow, examples, apache-kafka ]
+featured: true
+author: vjuranek
+---
+
+With the recent success of ChatGTP we can observe another wave of interest in the AI field and in machine learning in general.
+The previous wave of interest in this field was, at least to a certain extent, caused by the fact that awesome ML frameworks like https://www.tensorflow.org/[TensorFlow], https://pytorch.org/[PyTorch] or general data processing frameworks like https://spark.apache.org/[Spark] became available and made writing of ML models much more easy.
+Since that time these frameworks have matured and writing models is even easier, as you will see later in this blog.
+However, data set preparation and gathering data from various sources can be sometimes hard or even painful.
+Creating a complete pipeline which would pull existing or newly created data, adjust the data and ingest it into selected ML libraries can be quite a challenging task.
+Let's try to investigate if Debezium can help with this task and explore how we can leverage Debezium capabilities to make it more easy.
+
++++<!-- more -->+++
+
+== Change data capture and Debezium in ML pipelines
+
+In the machine learning area change data capture (CDC) can be a very powerful concept especially when it comes to online machine learning.
+However, CDC can also be an important part of the pipeline using pre-trained models. 
+We can use CDC to deliver new data immediately into a pre-trained model, which can evaluate it  and other parts of the pipeline can take any action based on the model output in the real time.
+
+Besides these use cases, Debezium obviously seems to be a very good fit for any pipeline which includes loading data from databases.
+Debezium can capture existing data as well as stream any newly created data.
+Another important feature of Debezium is support for single message transforms.
+We can adjust the data at the very beginning of the whole pipeline.
+Being it either some data transformation or filtering, we can send over the wire only data we are really interested in and thus eventually save the bandwidth and speed up the whole pipeline.
+Last, but not least, Debezium is able to deliver records into several message brokers and more brokers are being added (several new ones are available in recent 2.2.0 release).
+This increases the chances to integrate Debezium with other tools and add it into most of the data pipelines.
+In the future, Debezium potential can be increased further, e.g. by extending CDC not only to databases.
+
+So, this is the theory.
+Now let's explore how it works in reality.
+In this blog post we will look at how to stream data into TensorFlow.
+Based on the interest from the community, this may result in a series of blog posts where we try to explore possible integrations with other ML libraries and frameworks.
```

Comment:
```suggestion
This blog post will look at how to stream data into TensorFlow.
Based on the interest from the community, this may result in a series of blog posts where we explore possible integrations with other ML libraries and frameworks.
```